controllers/drplacementcontrol_controller.go

```diff
@@ -710,22 +712,56 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	ensureDRPCConditionsInited(&drpc.Status.Conditions, drpc.Generation, "Initialization")
 
-	placementObj, err := getPlacementOrPlacementRule(ctx, r.Client, drpc, logger)
-	if err != nil && !(errors.IsNotFound(err) && rmnutil.ResourceIsDeleted(drpc)) {
+	if rmnutil.ResourceIsDeleted(drpc) {
+		err := r.processDeletion(ctx, drpc, logger)
+		if err != nil {
+			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
+
+			statusErr := r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
+			if statusErr != nil {
+				err = fmt.Errorf("drpc deletion failed: %w and status update failed: %w", err, statusErr)
+			}
+
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	_, ramenConfig, err := ConfigMapGet(ctx, r.APIReader)
```

Comment:
I am not sure I understand the rationale behind separating the processing of DRPC and Placement deletion, especially considering that the `ramenConfig` variable isn't used until the `createDRPCInstance` call. This separation appears to introduce unnecessary duplication without clear benefits.